akka-actor-tests/src/test/scala/akka/actor/CoordinatedShutdownSpec.scala

```diff
@@ -449,6 +450,82 @@ class CoordinatedShutdownSpec
       }
     }
 
+    def withCoordinatedShutdown(phases: Map[String, Phase])(block: CoordinatedShutdown => Unit): Unit = {
+      val co = new CoordinatedShutdown(extSys, phases)
+      try {
+        block(co)
+      } finally {
+        co.terminationWatcher ! PoisonPill
+      }
+    }
+
+    "support actor termination tasks with a stop message" in {
+      val phases = Map("a" -> abortingPhase)
+      withCoordinatedShutdown(phases) { co =>
+        val actorToWatch = TestProbe()
+        co.addActorTerminationTask("a", "a1", actorToWatch.ref, Some("stop"))
+        val result = co.run(UnknownReason)
+        actorToWatch.expectMsg("stop")
+        result.isReadyWithin(1.second) should be(false)
+        actorToWatch.ref ! PoisonPill
+        result.futureValue should ===(Done)
+      }
+    }
+
+    "support actor termination tasks without a stop message" in {
+      val phases = Map("a" -> abortingPhase)
+      withCoordinatedShutdown(phases) { co =>
+        val actorToWatch = TestProbe()
+        co.addActorTerminationTask("a", "a1", actorToWatch.ref, None)
+        val result = co.run(UnknownReason)
+        actorToWatch.expectNoMessage()
```

Comment:
always use a short timeout for `expectNoMessage`, otherwise it will default to "within" (3 seconds)
100.millis would be enough here